core/dbt/parser/generic_test_builders.py

```diff
@@ -226,6 +226,12 @@ def __init__(
         self.namespace: str = groups['test_namespace']
         self.config: Dict[str, Any] = {}
 
+        # This code removes keys identified as config args from the test entry
+        # dictionary. The keys remaining in the 'args' dictionary will be
+        # "kwargs", or keyword args that are passed to the test macro.
+        # The "kwargs" are not rendered into strings until compilation time.
+        # The "configs" are rendered here (since they were not rendered back
+        # in the 'get_key_dicts' methods in the schema parsers).
```

Comment:
Thanks for the comment here. This is quite confusing behavior, and we should aim to eventually refactor it. I just realized one more way it can confuse: test kwargs _are_ included when creating the test node's name and `unique_id`, which are created at parse time, even though they're not _really_ rendered until execution time.

There's some conceptual merit to this distinction. A test with different kwargs is a _different test_ (e.g. `accepted_values` checking for `1,2,3` instead of `a,b,c`), whereas a test with different configs is just the same test being configured differently. But this is a fuzzy boundary at best, with some odd consequences. I think we should spend more time rethinking it after v1.